crates/bitwarden-vault/src/cipher/cipher_client/delete_attachment.rs

```diff
@@ -0,0 +1,453 @@
+use bitwarden_api_api::{apis::ciphers_api::CiphersApi, models};
+use bitwarden_core::require;
+use bitwarden_state::repository::Repository;
+#[cfg(feature = "wasm")]
+use wasm_bindgen::prelude::wasm_bindgen;
+
+use crate::{Cipher, CipherError, CipherId, CiphersClient, VaultParseError};
+
+/// Standalone function to delete an attachment from a cipher that is extracted for ease of unit
+/// testing.
+async fn delete_attachment(
+    api_client: &dyn CiphersApi,
+    repository: &dyn Repository<Cipher>,
+    cipher_id: CipherId,
+    attachment_id: &str,
+    admin: bool,
+) -> Result<Cipher, CipherError> {
+    let response = if admin {
+        api_client
+            .delete_attachment_admin(cipher_id.into(), attachment_id)
+            .await?
+    } else {
+        api_client
+            .delete_attachment(cipher_id.into(), attachment_id)
+            .await?
+    };
+
+    let boxed_cipher: Box<models::Cipher> = require!(response.cipher);
+    let cipher_response = *boxed_cipher;
+    let mut cipher = require!(repository.get(cipher_id.to_string()).await?);
+
+    cipher.revision_date = require!(cipher_response.revision_date)
```

Comment:
Nit: You don't need to manually deref the cipher to unbox it in this case:

```suggestion
    let cipher_response: Box<models::Cipher> = require!(response.cipher);
    let mut cipher = require!(repository.get(cipher_id.to_string()).await?);

    cipher.revision_date = require!(cipher_response.revision_date)
```

Reply:
Thank you for the code suggestion @dani-garcia!
First _real_ Rust work had me kinda going around in circles with my borrowing 😵‍💫

Reply:
Resolved ✅